SearchEngine/lambda_SearchEngine/include/KeyRecommander.h

```diff
@@ -0,0 +1,70 @@
+#ifndef __KEYRECOMMANDER_H__
+#define __KEYRECOMMANDER_H__
+
+#include "Server_TcpConnection.h"
+#include "CnDictionary.h"
+#include "EngDictionary.h"
+#include "EditDistance.h"
+#include <queue>
+#include <set>
+#include <string>
+#include <memory>
+#include <iostream>
+
+struct MyResult
+{
+    string word;
+    int frequency;
+    int distance;
+};
+
+struct MyCompare
+{
+    bool operator()(const MyResult & lhs,const MyResult & rhs) const
+    {
+        if (lhs.distance != rhs.distance) 
+        {
+            return lhs.distance > rhs.distance;
+        }
+        else if (lhs.frequency != rhs.frequency) 
+        {
```

Comment:
remove else to make it clearer.
the else here is duplicated since we have a `return` in line 27 which mean if we can reach line 29, then 
definitely `lhs.distance != rhs.distance`